R/summarise_draws.R

```diff
@@ -60,14 +64,66 @@ summarize_draws <- summarise_draws
 
 
 #' @export
-summarise_draws.default <- function(x, ...) {
-  x <- as_draws(x)
-  summarise_draws(x, ...)
+summarise_draws.default <- function(.x, ...) {
+  .x <- as_draws(.x)
+  summarise_draws(.x, ...)
+}
+
+# Helper functions
+create_summary_list <- function(.x, v, funs, .args) {
+  draws <- drop_dims_or_classes(.x[, , v], dims = 3, reset_class = FALSE)
+  args <- c(list(draws), .args)
+  v_summary <- named_list(names(funs))
+  for (m in names(funs)) {
+    v_summary[[m]] <- do.call(funs[[m]], args)
+  }
+  v_summary
+}
+
+summarise_draws_helper <- function(.x, funs, .args) {
+  variables_x <- variables(.x)
+  # get length and output names, calculated on the first variable
+  out1 <- create_summary_list(.x, variables_x[1], funs, .args)
```

Comment:
here there's `out1` (no underscore) but just above there's `variables_x` (with underscore). Either convention is fine but I would pick one or the other.